docs/usecase.rst

```diff
@@ -37,14 +72,37 @@ compute/data/communications abstractions,
 quality of service guarantees,
 and performance optimizations.
 
+.. seealso:: :doc:`executor`
+
 Generic sequential task execution
 ---------------------------------
 
 Units of work (nodes) describe the inputs and expected outputs,
 and identify a module able to perform the operation.
```

Comment:
What 'module' means in this context?

Reply:
Module meaning a software component. An importable Python module (for a Python-based execution Context), a (as yet unspecified) registered C++ functor, or some wrapper. The gist is that operations are discoverable with standardized syntax and semantics.

Reply:
Thanks. This means then that each function the end user wants to execute needs to be packaged into one python module? Here what software component is doing the discovery?

Reply:
I have assumed (Option 1)
* packages that want to offer first-class support of our interface (gmxapi, for instance) may adopt a few conventions or apply a few decorators to provide a set of compliant interfaces, and follow the convention of providing an importable namespace.operation helper function to implement namespace.operation work nodes.
* we can provide wrappers and helpers that allow easy construction of compliant operations for such things as command line tools
* users can apply a provided helper to arbitrary Python functions (like gmxapi `@function_wrapper` or Parsl `@App` decorator)

Alternative options include using Python package metadata to register SCALE-MS plugin entry points, but the main Option 2 is to find a syntax by which the scripting package can try to chase down operation details, and/or (Option 3) to provide a bunch of stub adapters for tools in scale-ms subpackages.